index.js

```diff
@@ -0,0 +1,209 @@
+const typeScriptSettings = {
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    sourceType: 'module',
+    project: './tsconfig.json',
+  },
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
+  ],
+  rules: {
+    '@typescript-eslint/array-type': ['error', { default: 'generic', readonly: 'generic' }],
+    '@typescript-eslint/ban-ts-comment': 'error',
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/explicit-member-accessibility': 'error',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/naming-convention': [
```

Comment:
Should we allow `UPPER_CASE` for `property` selector as well?

In APIs, enum-like values are often returned as UPPER_CASE values, where FE uses that as mapping, creating objects like:
```ts
  const keyToString: Record<string, string> = {
    EMAIL_ALREADY_EXIST: 'This email is already registered.',
  };
```
The above is currently not allowed, and the workaround is to add a `/* eslint-disable @typescript-eslint/naming-convention */` (+ disable) around the object.

Other use cases are `RouteNames`, `Params` and similar enum-line const-objects that we use for configuration keys. These are all written in the form of:

```js
const Params = {
  ID: 'id',
  TYPE: 'type',
};
console.log(Param.TYPE);
```

The current rule will fail on those as well.

As a reference, for actual TS `enums`, we follow a different standard:
```ts
// INCORRECT
export enum TimeDensity {
  HALF_HOUR = 200,
  HOUR = 100,
}
console.log(TimeDensity.HALF_HOUR);

// CORRECT
export enum TimeDensity {
  HalfHour = 200,
  Hour = 100,
}
console.log(TimeDensity.HalfHour);
```

Since enums are not supposed to have custom values (although they technically can, bot strings and numbers), we might need to decide on how to use both the `obj` and `enum` versions, if their casing is different or should be similar.

Reply:
I think `UPPER_CASE` for `property` selector should not be the default. But if your project works with an API that uses that naming convention, you could enable it in your own project in the linting settings.

Reply:
What about the other case?

Reply:
For `RouteNames` and `Params` we might consider writing them as:
```
const PARAMS = {
  id: 'id',
  type: 'type',
};
console.log(PARAMS.type);
```
I am not sure if I like it, but this is correct according to (current) the linting rule configuration.

Reply:
You cannot compare a `const` with an `enum`, because an `enum` is also a type:
```
export enum TimeDensity {
  HalfHour = 200,
  Hour = 100,
}

const time: TimeDensity = TimeDensity.HalfHour;
```

Reply:
But in my recent projects I barely used enums and I think they are less needed now you can do things like:
```type Direction = 'left' | 'right'```

Reply:
And we might even do the same for `Params`:
```
type Params = 'id' | 'type';
```

Reply:
> You cannot compare a const with an enum, because an enum is also a type

You can compare the part that is the same though :) They serve the same purpose.
Using the value could be the same, but is written differently: `PARAMS.type` vs `Params.Type`.

My preference goes to `routeNames.ID`, although they are a cons together, the important parts are the values (not the object, that's just a container to move them around). So if you destructure them, they keep their upper-const-casing :)

You could even named-export them as separate const values from a single file, where they also have UPPER_CASE names.

Reply:
>  I barely used enums and I think they are less needed now you can do ....

Using string literal union types is fun, and fine for typing. But passing those values, and using your IDE to find where a certain value is passed is going to be hell. Especially if the same value can coincidentally be used in multiple different "enums".

Good luck searching for `'type'` in your complete codebase, or other values that are even more ambiguous. Using "find reference/usage" on const objects or enums is way easier. :)